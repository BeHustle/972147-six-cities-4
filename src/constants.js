const NEAR_PLACES_COUNT = 3;

const MAX_REVIEWS_COUNT = 10;

const DEFAULT_AVATAR = `img/avatar.svg`;

const ICON_SIZE = [30, 30];

const ICON_PATH = `/img/pin.svg`;
const ACTIVE_ICON_PATH = `/img/pin-active.svg`;

const Screen = {
  MAIN: `main`,
  OFFER: `offer`,
  LOGIN: `login`
};

const AppStatus = {
  LOADING: `LOADING`,
  SUCCESS_LOAD: `SUCCESS_LOAD`,
  FAIL_LOAD: `FAIL_LOAD`
};

const AuthStatus = {
  NO_AUTH: `NO_AUTH`,
  AUTH: `AUTH`
};

const ErrorCode = {
  UNAUTHORIZED: 401
};

const Sorts = {
  POPULAR: `Popular`,
  PRICE_ASC: `Price: low to high`,
  PRICE_DESC: `Price: high to low`,
  TOP_RATED: `Top rated first`,
};

const CARD_TYPE = {
  MAIN: `main`,
  CARD_DETAIL: `card_detail`
};

const ReviewLength = {
  MIN: 50,
  MAX: 300
};

const HouseType = [
  `apartment`,
  `room`,
  `house`,
  `hotel`
];

const CommentStatus = {
  SUCCESS: `SUCCESS`,
  FAIL: `FAIL`,
  NOT_SEND: `NOT_SEND`
};

const CommentMessage = {
  SUCCESS: `Your comment has been successfully added. Wait moderation`,
  ERROR: `Error while adding a comment. Check your comment and try again`
};

export {
  NEAR_PLACES_COUNT,
  MAX_REVIEWS_COUNT,
  ICON_SIZE,
  ICON_PATH,
  ACTIVE_ICON_PATH,
  DEFAULT_AVATAR,
  Screen,
  Sorts,
  CARD_TYPE,
  AppStatus,
  HouseType,
  AuthStatus,
  ErrorCode,
  ReviewLength,
  CommentStatus,
  CommentMessage
};
