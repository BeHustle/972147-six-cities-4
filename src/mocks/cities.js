export const cities = [
  {
    id: 1,
    name: `Amsterdam`,
    coordinates: [52.38333, 4.9]
  },
  {
    id: 2,
    name: `Paris`,
    coordinates: [52.38333, 4.9]
  },
  {
    id: 3,
    name: `Cologne`,
    coordinates: [52.38333, 4.9]
  },
  {
    id: 4,
    name: `Brussels`,
    coordinates: [52.38333, 4.9]
  },
  {
    id: 5,
    name: `Hamburg`,
    coordinates: [52.38333, 4.9]
  },
  {
    id: 6,
    name: `Dusseldorf`,
    coordinates: [52.38333, 4.9]
  },
];
