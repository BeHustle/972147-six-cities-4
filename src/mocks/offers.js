export const offers = [
  {
    id: 1,
    name: `Beautiful & luxurious apartment at great location`,
    price: 132,
    image: `img/apartment-01.jpg`,
    type: `Private room`,
    isPremium: false,
    inBookmarks: true,
    rating: 4
  },
  {
    id: 2,
    name: `Wood and stone place`,
    price: 254,
    image: `img/apartment-02.jpg`,
    type: `Apartment`,
    isPremium: false,
    inBookmarks: false,
    rating: 5
  },
  {
    id: 3,
    name: `Canal View Prinsengracht`,
    price: 178,
    image: `img/apartment-03.jpg`,
    type: `Private room`,
    isPremium: true,
    inBookmarks: true,
    rating: 3
  },
  {
    id: 4,
    name: `Nice, cozy, warm big bed apartment`,
    price: 995,
    image: `img/room.jpg`,
    type: `Apartment`,
    isPremium: true,
    inBookmarks: false,
    rating: 1
  },
  {
    id: 5,
    name: `Beautiful & luxurious apartment at great location`,
    price: 80,
    image: `img/apartment-01.jpg`,
    type: `Private room`,
    isPremium: false,
    inBookmarks: true,
    rating: 2
  },
];
