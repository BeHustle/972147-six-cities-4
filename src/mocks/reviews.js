export const reviews = [
  {
    id: 1,
    offerId: 1,
    author: {
      name: `Max`,
      avatar: `img/avatar-angelina.jpg`
    },
    text: `Lorem ipsum dolor sit amet, consectetur adipisicing elit. Minima, quis!`,
    rating: 5,
    date: new Date(`2019-12-17`)
  },
  {
    id: 2,
    offerId: 1,
    author: {
      name: `Alex`,
      avatar: `img/avatar-max.jpg`
    },
    text: `Lorem ipsum dolor sit amet!`,
    rating: 3,
    date: new Date(`2020-07-17`)
  },
  {
    id: 3,
    offerId: 3,
    author: {
      name: `William`,
      avatar: ``
    },
    text: `Lorem ipsum dolor sit amet, consectetur adipisicing elit. At ipsa labore libero nam sunt temporibus?`,
    rating: 1,
    date: new Date(`2018-03-25`)
  },
  {
    id: 4,
    offerId: 2,
    author: {
      name: `Lucy`,
      avatar: `img/avatar-angelina.jpg`
    },
    text: `Lorem ipsum dolor sit amet, consectetur adipisicing elit.`,
    rating: 4,
    date: new Date(`2020-01-13`)
  },
];
