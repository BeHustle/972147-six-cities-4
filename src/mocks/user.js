const email = `Oliver.conner@gmail.com`;

export {email};
