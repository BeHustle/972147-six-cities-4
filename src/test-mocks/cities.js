export const cities = [
  {
    id: 1,
    name: `Amsterdam`,
    coordinates: [52.38333, 4.9],
    zoom: 10
  },
  {
    id: 2,
    name: `Paris`,
    coordinates: [48.85875194, 2.33926391],
    zoom: 10
  },
  {
    id: 3,
    name: `Cologne`,
    coordinates: [50.938361, 6.959974],
    zoom: 10
  },
  {
    id: 4,
    name: `Brussels`,
    coordinates: [50.8465573, 4.351697],
    zoom: 10
  },
  {
    id: 5,
    name: `Hamburg`,
    coordinates: [53.550341, 10.000654],
    zoom: 10
  },
  {
    id: 6,
    name: `Dusseldorf`,
    coordinates: [51.2254018, 6.7763137],
    zoom: 10
  },
];
