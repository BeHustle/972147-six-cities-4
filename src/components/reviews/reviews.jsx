import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {AuthStatus} from '../../constants.js';
import {Operation as DataOperation} from '../../reducer/data/data.reducer.js';
import {getReviews} from '../../reducer/data/data.selectors.js';
import {getAuthStatus} from '../../reducer/user/user.selectors.js';
import AddReview from '../add-review/add-review.jsx';
import Review from '../review/review.jsx';

class Reviews extends React.PureComponent {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    const {onReviewsMount, offerId} = this.props;
    onReviewsMount(offerId);
  }

  render() {
    const {reviews, authStatus, offerId} = this.props;
    return <section className="property__reviews reviews">
      <h2 className="reviews__title">Reviews &middot; <span className="reviews__amount">{reviews.length}</span></h2>
      <ul className="reviews__list">
        {reviews.map((review) =>
          <Review key={review.id} review={review}/>,
        )}
      </ul>
      {authStatus === AuthStatus.AUTH && <AddReview offerId={offerId} />}
    </section>;
  }
}

Reviews.propTypes = {
  offerId: PropTypes.number.isRequired,
  reviews: PropTypes.arrayOf(PropTypes.exact({
    id: PropTypes.number.isRequired,
    author: PropTypes.exact({
      name: PropTypes.string.isRequired,
      avatar: PropTypes.string,
    }),
    text: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    date: PropTypes.instanceOf(Date),
  })),
  onReviewsMount: PropTypes.func.isRequired,
  authStatus: PropTypes.oneOf(Object.values(AuthStatus)),
};

const mapDispatchToProps = (dispatch) => ({
  onReviewsMount(offerId) {
    dispatch(DataOperation.loadReviews(offerId));
  },
});

const mapStateToProps = (state) => ({
  reviews: getReviews(state),
  authStatus: getAuthStatus(state)
});

export default connect(mapStateToProps, mapDispatchToProps)(Reviews);
