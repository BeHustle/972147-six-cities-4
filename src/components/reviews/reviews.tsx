import * as React from 'react';
import {connect} from 'react-redux';
import {AuthStatus} from '../../constants';
import {Operation as DataOperation} from '../../reducer/data/data.reducer';
import {getReviews} from '../../reducer/data/data.selectors';
import {getAuthStatus} from '../../reducer/user/user.selectors';
import AddReview from '../add-review/add-review';
import Review from '../review/review';
import {ReviewInterface} from "../../types";

interface Props {
  offerId: number;
  onReviewsMount: (offerId: number) => void;
  reviews: Array<ReviewInterface>;
  authStatus: string;
}

class Reviews extends React.PureComponent<Props, {}> {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    const {onReviewsMount, offerId} = this.props;
    onReviewsMount(offerId);
  }

  render() {
    const {reviews, authStatus, offerId} = this.props;
    return <section className="property__reviews reviews">
      <h2 className="reviews__title">Reviews &middot; <span className="reviews__amount">{reviews.length}</span></h2>
      <ul className="reviews__list">
        {reviews.map((review) =>
          <Review key={review.id} review={review}/>,
        )}
      </ul>
      {authStatus === AuthStatus.AUTH && <AddReview offerId={offerId} />}
    </section>;
  }
}

const mapDispatchToProps = (dispatch) => ({
  onReviewsMount(offerId) {
    dispatch(DataOperation.loadReviews(offerId));
  },
});

const mapStateToProps = (state) => ({
  reviews: getReviews(state),
  authStatus: getAuthStatus(state)
});

export {Reviews};

export default connect(mapStateToProps, mapDispatchToProps)(Reviews);
