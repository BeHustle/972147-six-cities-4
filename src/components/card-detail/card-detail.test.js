import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Sorts} from '../../constants.js';
import CardDetail from './card-detail.jsx';
import {cities} from '../../test-mocks/cities.js';
import {offers} from '../../test-mocks/offers.js';
import {email} from '../../test-mocks/user.js';
import {reviews} from '../../test-mocks/reviews.js';

const mockStore = configureStore([]);
jest.mock(`../map/map.jsx`, () => `map`);

it(`Card detail render`, () => {
  const store = mockStore({
    offers,
    city: cities[0],
    cities,
    userEmail: email,
    sorts: Sorts,
    activeSort: Sorts.POPULAR,
    activeOfferId: null,
    reviews
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <CardDetail
            offerId={1}
            onCardTitleClick={() => {}}
          />
        </Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
