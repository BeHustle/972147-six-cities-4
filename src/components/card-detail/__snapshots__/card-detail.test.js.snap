// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card detail render 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  user@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Beautiful & luxurious apartment at great locatccion"
              className="property__image"
              src="img/room.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Beautiful & luxurious apartment at great locatccion"
              className="property__image"
              src="img/apartment-01.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Beautiful & luxurious apartment at great locatccion"
              className="property__image"
              src="img/apartment-02.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Beautiful & luxurious apartment at great locatccion"
              className="property__image"
              src="img/apartment-03.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Beautiful & luxurious apartment at great locatccion"
              className="property__image"
              src="img/studio-01.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Beautiful & luxurious apartment at great locatccion"
              className="property__image"
              src="img/apartment-01.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              Beautiful & luxurious apartment at great locatccion
            </h1>
            <button
              className="property__bookmark-button button property__bookmark-button--active"
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height="33"
                width="31"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "100%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              4.1
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              hotel
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              3 Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 4 adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              132
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                Wi-Fi
              </li>
              <li
                className="property__inside-item"
              >
                Washing machine
              </li>
              <li
                className="property__inside-item"
              >
                Towels
              </li>
              <li
                className="property__inside-item"
              >
                Heating
              </li>
              <li
                className="property__inside-item"
              >
                Coffee machine
              </li>
              <li
                className="property__inside-item"
              >
                Baby seat
              </li>
              <li
                className="property__inside-item"
              >
                Kitchen
              </li>
              <li
                className="property__inside-item"
              >
                Dishwasher
              </li>
              <li
                className="property__inside-item"
              >
                Cabel TV
              </li>
              <li
                className="property__inside-item"
              >
                Fridge
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper user__avatar-wrapper property__avatar-wrapper--pro"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height="74"
                  src="img/avatar-angelina.jpg"
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                Angelina
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
      The building is green and from 18th century.
              </p>
              <p
                className="property__text"
              >
                An independent House, strategically located between Rembrand Square and National Opera, but where the
      bustle of the city comes to rest in this alley flowery and colorful.
              </p>
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <h2
              className="reviews__title"
            >
              Reviews · 
              <span
                className="reviews__amount"
              >
                4
              </span>
            </h2>
            <ul
              className="reviews__list"
            >
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="William"
                      className="reviews__avatar user__avatar"
                      height="54"
                      src="img/avatar.svg"
                      width="54"
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    William
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "20%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    Lorem ipsum dolor sit amet, consectetur adipisicing elit. At ipsa labore libero nam sunt temporibus?
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2018-03-25"
                  >
                    March 2018
                  </time>
                </div>
              </li>
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Max"
                      className="reviews__avatar user__avatar"
                      height="54"
                      src="img/avatar-angelina.jpg"
                      width="54"
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Max
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "100%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    Lorem ipsum dolor sit amet, consectetur adipisicing elit. Minima, quis!
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2019-12-17"
                  >
                    December 2019
                  </time>
                </div>
              </li>
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Lucy"
                      className="reviews__avatar user__avatar"
                      height="54"
                      src="img/avatar-angelina.jpg"
                      width="54"
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Lucy
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    Lorem ipsum dolor sit amet, consectetur adipisicing elit.
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2020-01-13"
                  >
                    January 2020
                  </time>
                </div>
              </li>
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Alex"
                      className="reviews__avatar user__avatar"
                      height="54"
                      src="img/avatar-max.jpg"
                      width="54"
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Alex
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "60%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    Lorem ipsum dolor sit amet!
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2020-07-17"
                  >
                    July 2020
                  </time>
                </div>
              </li>
            </ul>
            <form
              action="#"
              className="reviews__form form"
              method="post"
              onSubmit={[Function]}
            >
              <label
                className="reviews__label form__label"
                htmlFor="review"
              >
                Your review
              </label>
              <div
                className="reviews__rating-form form__rating"
              >
                <input
                  className="form__rating-input visually-hidden"
                  id="5-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                  value="5"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="5-stars"
                  title="perfect"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  id="4-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                  value="4"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="4-stars"
                  title="good"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  id="3-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                  value="3"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="3-stars"
                  title="not bad"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  id="2-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                  value="2"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="2-stars"
                  title="badly"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  id="1-star"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                  value="1"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="1-star"
                  title="terribly"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
              </div>
              <textarea
                className="reviews__textarea form__textarea"
                id="review"
                name="review"
                onChange={[Function]}
                placeholder="Tell how was your stay, what you like and what can be improved"
              />
              <div
                style={
                  Object {
                    "display": "none",
                    "fontWeight": 500,
                    "marginBottom": "10px",
                    "marginTop": "10px",
                    "textAlign": "right",
                  }
                }
              />
              <div
                className="reviews__button-wrapper"
              >
                <p
                  className="reviews__help"
                >
                  To submit review please make sure to set 
                  <span
                    className="reviews__star"
                  >
                    rating
                  </span>
                   and describe your stay with at least 
                  <b
                    className="reviews__text-amount"
                  >
                    50 characters
                  </b>
                  .
                </p>
                <button
                  className="reviews__submit form__submit button"
                  disabled={true}
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </form>
          </section>
        </div>
      </div>
      <map
        coordinates={
          Array [
            52.3909553943508,
            4.85309666406198,
          ]
        }
        offers={
          Array [
            Object {
              "author": Object {
                "avatar": "img/avatar-angelina.jpg",
                "isSuper": true,
                "name": "Angelina",
              },
              "cityId": 1,
              "coordinates": Array [
                52.3909553943508,
                4.85309666406198,
              ],
              "facilities": Array [
                "Wi-Fi",
                "Washing machine",
                "Towels",
                "Heating",
                "Coffee machine",
                "Baby seat",
                "Kitchen",
                "Dishwasher",
                "Cabel TV",
                "Fridge",
              ],
              "guests": "Max 4 adults",
              "id": 1,
              "image": "img/apartment-01.jpg",
              "images": Array [
                "img/room.jpg",
                "img/apartment-01.jpg",
                "img/apartment-02.jpg",
                "img/apartment-03.jpg",
                "img/studio-01.jpg",
                "img/apartment-01.jpg",
              ],
              "inBookmarks": true,
              "isPremium": true,
              "name": "Beautiful & luxurious apartment at great locatccion",
              "price": 132,
              "rating": 4.1,
              "rooms": "3 Bedrooms",
              "text": Array [
                "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
      The building is green and from 18th century.",
                "An independent House, strategically located between Rembrand Square and National Opera, but where the
      bustle of the city comes to rest in this alley flowery and colorful.",
              ],
              "type": "hotel",
              "zoom": 10,
            },
            Object {
              "author": Object {
                "avatar": "img/avatar-max.jpg",
                "isSuper": true,
                "name": "Max",
              },
              "cityId": 1,
              "coordinates": Array [
                52.369553943508,
                4.85309666406198,
              ],
              "facilities": Array [
                "Wi-Fi",
                "Washing machine",
                "Towels",
              ],
              "guests": "Max 2 adults",
              "id": 2,
              "image": "img/apartment-02.jpg",
              "images": Array [
                "img/room.jpg",
                "img/apartment-01.jpg",
                "img/apartment-02.jpg",
                "img/apartment-03.jpg",
                "img/studio-01.jpg",
                "img/apartment-01.jpg",
              ],
              "inBookmarks": false,
              "isPremium": false,
              "name": "Wood and stone place",
              "price": 254,
              "rating": 5,
              "rooms": "1 Bedrooms",
              "text": Array [
                "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
      The building is green and from 18th century.",
                "An independent House, strategically located between Rembrand Square and National Opera, but where the
      bustle of the city comes to rest in this alley flowery and colorful.",
              ],
              "type": "apartment",
              "zoom": 10,
            },
            Object {
              "author": Object {
                "avatar": "",
                "isSuper": true,
                "name": "Alex",
              },
              "cityId": 2,
              "coordinates": Array [
                48.85875194,
                2.33926391,
              ],
              "facilities": Array [
                "Wi-Fi",
                "Heating",
                "Coffee machine",
                "Baby seat",
                "Kitchen",
              ],
              "guests": "Max 2 adults and 1 children",
              "id": 3,
              "image": "img/apartment-03.jpg",
              "images": Array [
                "img/room.jpg",
                "img/apartment-01.jpg",
                "img/apartment-02.jpg",
                "img/apartment-03.jpg",
                "img/studio-01.jpg",
                "img/apartment-01.jpg",
              ],
              "inBookmarks": true,
              "isPremium": true,
              "name": "Canal View Prinsengracht",
              "price": 178,
              "rating": 3,
              "rooms": "2 Bedrooms",
              "text": Array [
                "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
      The building is green and from 18th century.",
                "An independent House, strategically located between Rembrand Square and National Opera, but where the
      bustle of the city comes to rest in this alley flowery and colorful.",
              ],
              "type": "room",
              "zoom": 10,
            },
          ]
        }
        type="card_detail"
        zoom={10}
      />
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <div
          className="places__list near-places__list"
        >
          <article
            className="place-card near-places__card"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="place-card__mark"
            >
              <span>
                Premium
              </span>
            </div>
            <div
              className="place-card__image-wrapper near-places__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="200"
                  src="img/apartment-01.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    132
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button place-card__bookmark-button--active"
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "100%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  Beautiful & luxurious apartment at great locatccion
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                hotel
              </p>
            </div>
          </article>
          <article
            className="place-card near-places__card"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="place-card__image-wrapper near-places__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="200"
                  src="img/apartment-02.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    254
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button false"
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "100%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  Wood and stone place
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                apartment
              </p>
            </div>
          </article>
          <article
            className="place-card near-places__card"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="place-card__mark"
            >
              <span>
                Premium
              </span>
            </div>
            <div
              className="place-card__image-wrapper near-places__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="200"
                  src="img/apartment-03.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    178
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button place-card__bookmark-button--active"
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "60%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  Canal View Prinsengracht
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                room
              </p>
            </div>
          </article>
        </div>
      </section>
    </div>
  </main>
</div>
`;
